app/helpers/office_helper.rb

```diff
@@ -10,6 +12,12 @@ def office_address(office)
   end
 
   def display_website(office, firm)
-    office.website.present? ? office.website : firm.website_address
+    user_uri = office.website.present? ? office.website : firm.website_address
+    uri = URI(user_uri.strip)
+    "#{uri.host}#{uri.path}"
+  end
+
+  def website_url(office, firm)
+    "#{SCHEME}#{display_website(office, firm)}"
```

Comment:
Is `#display_website` helper method called anywhere else? 
If not, have you considered to put everything together in a single method (`website_url` seems better name) as is not particularly long/complex and we only extracted the latest the URL build, that is just a concatenation in the last line?
So doing this `"#{SCHEME}#{uri.host}#{uri.path}"` instead of this `"#{uri.host}#{uri.path}"` we saved a method/abstraction/indirection.

Another option would be to leave the `display_website` as it is and leave all the responsibility for the URL sanitation + build for `website_url`, but TBH I would just put everything together (its a 3 lines method).

Reply:
Ok, let me dig a bit deeper and see how's been used (I have seen at least another call)

Reply:
I'm wondering if it's a naming thing? Until I read the code, I wasn't sure what a `display_website` was.  Whilst I can see the argument for making it 2 methods, I'd be inclined to make it one method partly because I can't think of a better name for `display_website`. The best I could come up with was `schemeless_url` 😄  

I'd also be inclined to name the method `office_website`. In the view context, `link_to website_url` is a little redundant and `link_to office_website` is more declarative.